contracts/ethereum/l1Aggregator.sol

```diff
@@ -13,13 +13,15 @@ contract L1Aggregator {
     //TODO: an asset can be present on multiple Starknet bridges,
     //so approving all bridges where the token is collateralized should be considered in the future
     mapping(address => address) public bridges;
+    mapping(address => mapping(uint => bool)) public assetPublishedForPeriod;
```

Comment:
let's use the following mapping to cache the balance of each asset `mapping(address => mapping(uint => uint)) public assetPublishedForPeriod;`